### ISSUE TITLE
[select] mult-select variation doesn't allow for keyboard access

### ISSUE THREAD

**Issue by scottaohara** [openui/open-ui#476]
The 'mult-select' variation of the [explainer demos](https://patrickbrosset.com/lab/2022-01-06-selectmenu/) does not presently allow for keyboard focus.  Navigating using <kbd>tab</kbd> key, it is skipped over.

If one were to open it with the mouse, arrow keys can be used to navigate through the listbox popup's options.  However if <kbd>Space</kbd> is used, no selection occurs. If <kbd>Enter</kbd> key is used the listbox dismisses with no selection taking place.

When a selection is made by a mouse, a list is populated within the 'combobox' where the list item has a 'x' icon.  However the combobox remains inaccessible to keyboard and the selected option too is not reachable, nor is communicated as an interactive element which would indicate that it can be removed.  

Testing with screen readers, JAWS and NVDA skip over the component completely when navigating with scan mode.  Narrator can find the combobox, but always announces it in an expanded state, even with the popup isn't visible.  Unlike other screen readers, which skip over the combobox, Narrator can invoke the popup with the <kbd>Space</kbd> key.  but it cannot select an option, nor can it remove a selected option if one were chosen via mouse click.

Some of what might be needed to correct this example would need to touch on #458, and have clear guidance on what can be slotted for the element that invokes the popup.  E.g., a combobox > list pattern like this might need some revisions... and if someone were to replace the standard 'button', then it would have to be with a control that is keyboard focusable.

**Comment by dandclark** [comment]
> Some of what might be needed to correct this example would need to touch on https://github.com/openui/open-ui/issues/458, and have clear guidance on what can be slotted for the element that invokes the popup. E.g., a combobox > list pattern like this might need some revisions... and if someone were to replace the standard 'button', then it would have to be with a control that is keyboard focusable.

Another possibility might be to automatically make the element with `behavior="button"` focusable by default. That is, for cases like this:
```html
    <selectmenu class="multiselect">
      <div slot="button" behavior="button">
        <ul class="selected-values"></ul>
      </div>
      <option value="Broccolini">Broccolini</option>
     <option>...</option>
    </selectmenu>
```
We could do either of the following:
1) Treat the `<div>` as if it had `tabindex="0"`
2) Actually set `tabindex="0"` on the `<div>`.

We already give the div an implicit `role="combobox"`, so this would be a similar idea.

**Comment by scottaohara** [comment]
That may well work.  Another thing that we'd need to figure out too is exactly where the listing of selected items goes.  placing it in the combobox will mean that nvda/jaws will drop the list semantics and communicate the separate list items as just a single string of text.  if a dismiss button is added to each item choses, and given an accessible name such as "remove" to indicate it's purpose, then the value of the combobox would return as "item 1 remove item 2 remove item 3 remove etc.".  

however, if the list was populated as a sibling to the combobox control, and that consistently remained as a button (maybe displaying the number of items or 'multiple items selected'), then that could help mitigate the issue.

There are a few different ways that could be done.  but we can discuss that when we get more into the 'button' slotting.

**Comment by josepharhar** [comment]
I'd like to address any changes to the single-select selectmenu, since this issue seems to be mostly focused on multi-select selectmenu which we will be tackling later.

We could easily make the button part always keyboard focusable. @scottaohara does this sound like a good idea for single-select? Are there any other changes for single-select that you think we should do?

**Comment by josepharhar** [comment]
> We could easily make the button part always keyboard focusable.

This has effectively been done now that the button part is a `<button type=selectlist>`
